### DISCUSSION TITLE
Configured registry is not set in package-lock.json

### DISCUSSION THREAD

**Discussion by ikercrg** [renovatebot/renovate#8913] [Request Help]
**What Renovate type, platform and version are you using?**

I use the one which is explained here: https://gitlab.com/renovate-bot/renovate-runner

"renovateVersion": "24.63.0"

Our company has its own GitLab instance and I use a mirror of the renovate/runner project for creating the project that is mentioned.

**Describe the bug**

Having the following configuration in renovate.json:

`{
  "gradle-lite": {
    "enabled": true
  },
  "gradle": {
    "enabled": false
  },
  "registryUrls": [
    "https://myartifactory_url"
  ]
}`

After running the renovate job, in every dependency updated in the package-lock.json, I get something like this in the resolve part: 

`"resolved": "https://registry.npmjs.org/@angular/cdk/...` - NOT EXPECTED

instead of

`"resolved": "https://myartifactory_url/@angular/cdk/...` - EXPECTED

**Additional context**

How should it be the configuration, so I get only myartifactory_url in every change in the package-lock.json

**Comment by viceice** [comment]
you need to use a `packageRule` to assign `registryUrls`

and you need to add `npmrc:'registry=https://myartifactory_url'`

**Reply by ikercrg** [reply]
I updated to the following configuration:

```json
{
  "gradle-lite": {
    "enabled": true
  },
  "gradle": {
    "enabled": false
  },
  "packageRules": [
    {
      "matchDatasources": [
        "npm"
      ],
      "registryUrls": [
        "https://myartifactory_url"
      ]
    }
  ],
  "npmrc": "registry=https://myartifactory_url"
}
```

But I still have same problem. Should I set it another way or it is missing something?

**Reply by viceice** [reply]
do you have a `.npmrc` file in your repo? it shopuld contain `registry=https://myartifactory_url/` too.

Also i've seen that github npm proxy is redirecting packages to `https://registry.npmjs.org` so npm is writing them to the lockfile.
Maybe artifactory uses redirects too?

**Reply by ikercrg** [reply]
There is no .npmrc in our repo. We use the one in our machine for local development and for the pipeline, it is previously generated.

About redirection, I do not think so, because in every install, the dependencies are updated with our registry url

**Reply by ikercrg** [reply]
Solved! I created dynamically the .npmrc file and it works correctly.

Thanks for the answers
